lib/wam.coffee

```diff
@@ -237,23 +236,26 @@ downloadAddon = (addonURL, addonIdentifier) ->
     tmpFilePath = path.resolve os.tmpdir(), "#{addonIdentifier}-#{filename}.zip"
 
     log.info "Downloading #{addonIdentifier.magenta}..."
-
-    request {
-      followAllRedirects: true,
-      url: "#{addonURL}"
-    }
-    .pipe fs.createWriteStream "#{tmpFilePath}"
-    .on 'error', (error) ->
-      console.log error
-      if error.code is ErrorCodeDoesntExist
-        reject "Error writing Addon #{addonIdentifier.yellow} in temporary
-                folder #{tmpFilePath.yellow}."
-      else
+    
+    cs
+      method: 'GET'
+      uri: addonURL
+      rejectUnauthorized: false
+      realEncoding: null
+    .then (body) ->
+
+      buffer = Buffer.from body, 'utf8'
+
+      fs.writeFile tmpFilePath, buffer, (error) ->
+        if error
+          reject "Error saving Addon #{addonIdentifier.yellow}."
```

Comment:
As `cloudscraper` is returning a promise, there's no need to use `Bluebird` to wrap the request: it is a promise already

Here instad of calling reject you could throw an error, same thing in line 257. In line 253 you can just return `tmpFilePath`.